File: тесты/bench07_sum_squares.cpp
#include <iostream>
#include <chrono>

int main() {
    auto t0 = std::chrono::high_resolution_clock::now();
    long long sum = 0;
    long long mod = 1000000007;
    for (long long i = 1; i <= 10000000; ++i) {
        sum = (sum + i * i) % mod;
    }
    auto t1 = std::chrono::high_resolution_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    std::cout << "RESULT:" << sum << std::endl;
    std::cout << "TIME_MS:" << ms << std::endl;
    return 0;
}

File: тесты/bench07_sum_squares.py
import time

def compute_sum_squares(n):
    total = 0
    mod = 1000000007
    i = 1
    while i <= n:
        total = (total + i * i) % mod
        i += 1
    return total

if __name__ == "__main__":
    t0 = time.perf_counter()
    ans = compute_sum_squares(10000000)
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000.0
    print(f"RESULT:{ans}")
    print(f"TIME_MS:{ms:.2f}")
